lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -410,16 +412,24 @@ public function processParameterValue($value)
             return $value;
         }
 
-        if (is_object($value) && $this->_em->getMetadataFactory()->hasMetadataFor(ClassUtils::getClass($value))) {
+        if ($value instanceof Mapping\ClassMetadata) {
```

Comment:
Just found out via kcachegrind that this caused massive performance regressions when binding any non-scalar value as parameter.

For example:

```php
$thing = new DateTime();

$query->setParameter('foo', $thing);
$query->getResult();
```

Regardless of `$query`, this will lead to an internal class metadata lookup (and internal failure), plus exceptions being allocated and GC'd.

The following won't help either:

```php
$thing = new DateTime();

$query->setParameter('foo', $thing, DateTimeType::NAME);
$query->getResult();
```

That's because `processParameterValue($value)` is called even when the type is known.

@alcaeus @Ocramius would it be a good idea to skip the entire `processParameterValue()`when the type is explicitly passed in?

Reply:
Moved to https://github.com/doctrine/doctrine2/issues/7527